src/uwtools/drivers/driver.py

```diff
@@ -196,6 +165,133 @@ def _resources(self) -> Dict[str, Any]:
         """
         Returns configuration data for the runscript.
         """
+        try:
+            platform = self._config["platform"]
+        except KeyError as e:
+            raise UWConfigError("Required 'platform' block missing in config") from e
+        return {
+            "account": platform["account"],
+            "rundir": self._rundir,
+        }
+
+    @property
+    def _rundir(self) -> Path:
+        """
+        The path to the component's run directory.
+        """
+        return Path(self._driver_config["run_dir"])
+
+    def _taskname(self, suffix: str) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s" % (self._driver_name, suffix)
+
+    def _taskname_with_cycle(self, cycle: datetime, suffix: str) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s %s" % (cycle.strftime("%Y%m%d %HZ"), self._driver_name, suffix)
+
+    def _taskname_with_cycle_and_leadtime(
+        self, cycle: datetime, leadtime: timedelta, suffix: str
+    ) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s %s" % (
+            (cycle + leadtime).strftime("%Y%m%d %H:%M:%S"),
+            self._driver_name,
+            suffix,
+        )
+
+    def _validate(self) -> None:
+        """
+        Perform all necessary schema validation.
+        """
+        for schema_name in (self._driver_name.replace("_", "-"), "platform"):
+            validate_internal(schema_name=schema_name, config=self._config)
+
+
+class Driver(Assets):
+    """
+    An abstract class for standalone component drivers.
+    """
+
+    def __init__(
+        self,
+        config: Optional[Union[dict, Path]],
+        dry_run: bool = False,
+        batch: bool = False,
+        cycle: Optional[datetime] = None,
```

Comment:
I was wondering if changes from @maddenp-noaa #499 were added into the PR yet?

I noticed the ordering of the input parameters here and how he updated them in his pervious PR to have `cycle` and `leadtime` next to each other. Or possibly this just should be updated with the new `Driver` class. If so, I'll keep my eye out for where it appears in other places.
```
        cycle: Optional[datetime] = None,
        key_path: Optional[List[str]] = None,
        leadtime: Optional[timedelta] = None,
```

Reply:
This would be an error from resolving conflicts, thanks for keeping an eye out for this!